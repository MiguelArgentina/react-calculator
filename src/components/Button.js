import PropTypes from 'prop-types';

const Button = (props) => {
  const {
    name, clickHandler, color, wide,
  } = props;
  return (
    <button type="button" className={`${color} ${wide}`} onClick={() => clickHandler(name)}>
      { name }
    </button>
  );
};

Button.propTypes = {
  name: PropTypes.string.isRequired,
  color: PropTypes.string.isRequired,
  wide: PropTypes.string.isRequired,
  clickHandler: PropTypes.func.isRequired,
};

export default Button;
