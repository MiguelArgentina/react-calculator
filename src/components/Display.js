import PropTypes from 'prop-types';

const Display = (props) => {
  const { result } = props;
  return (
    <div className="border border-info bg-warning text-primary text-right px-1">
      { result }
    </div>
  );
};

Display.propTypes = {
  result: PropTypes.string,
};

Display.defaultProps = {
  result: '0',
};

export default Display;
