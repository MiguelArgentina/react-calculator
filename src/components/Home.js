const Home = () => (
  <div>
    <h1 className="text-start ms-4 mt-4 home-title"> Welcome to our page!</h1>
    <p className="text-start ms-4 mt-4 home-para">Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui o!cia deserunt mollit anim id est laborum. Ipsu</p>
    <p className="text-start ms-4 mt-4 home-para">Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui o!cia deserunt mollit anim id est laborum. Ipsu</p>
  </div>
);

export default Home;
