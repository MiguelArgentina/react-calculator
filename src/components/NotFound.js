import { Link } from 'react-router-dom';

const NotFound = () => (
  <>
    <div className="text-start home-para quote-para">
      The page you are looking for does not exist.
    </div>
    <p>
      <Link to="/" className="back-home-link">Back to home</Link>
    </p>
  </>
);

export default NotFound;
