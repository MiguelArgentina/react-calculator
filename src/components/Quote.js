import useFetch from './useFetch';

const Quote = () => {
  let data;
  let isPending = true;
  let error;

  if (isPending) ({ data, isPending, error } = useFetch('https://favqs.com/api/qotd'));

  return (
    <div>
      { isPending && <p className="text-start home-para quote-para">Loading...</p> }
      { error && <p className="text-start home-para quote-para">An error has occured while loading the quote...</p> }
      { data && (
      <p className="text-start home-para quote-para">
        {data.quote.body}
      </p>
      ) }
      { data && (
      <p className="text-start home-para quote-para">
        {data.quote.author}
      </p>
      ) }
    </div>
  );
};

export default Quote;
// GET https://api.paperquotes.com/apiv1/quotes/?lang=en/?tags=math,mathematics&order=-likes

// GET https://api.paperquotes.com/apiv1/quotes/?tags=love,motivation&order=-likes

// 1c2e6bad6065b2d3704818fdbd848de65f87b165
