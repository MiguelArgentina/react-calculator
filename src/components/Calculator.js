import { useState } from 'react';
import calculate from '../logic/calculate';
import ButtonPanel from './ButtonPanel';
import Display from './Display';

const Calculator = () => {
  const calcData = {
    total: null,
    next: '0',
    operation: null,
  };
  const [data, setData] = useState(calcData);
  const handleClick = (btnName) => setData((data) => calculate(data, btnName));
  const { next, total, operation } = data;
  return (
    <div className="mt-4d-block d-md-flex justify-content-between">
      <div className="calc-text my-3">LetÂ´s do some math!</div>
      <div className="calc-wrapper">
        <Display operation={operation} result={next || total} />
        <ButtonPanel clickHandler={handleClick} />
      </div>
    </div>
  );
};
export default Calculator;
