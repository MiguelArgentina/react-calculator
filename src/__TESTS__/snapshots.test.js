import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import Button from '../components/Button';
import Display from '../components/Display';
import NotFound from '../components/NotFound';
import Home from '../components/Home';
import Calculator from '../components/Calculator';
import Quote from '../components/Quote';

describe('Tests snapshots of components', () => {
  it('the Home component renders correctly:', () => {
    const homeSnapshot = renderer
      .create(
        <Home />,
      )
      .toJSON();
    expect(homeSnapshot).toMatchSnapshot();
  });

  it('the Calculator component renders correctly:', () => {
    const calcSnapshot = renderer
      .create(
        <Calculator />,
      )
      .toJSON();
    expect(calcSnapshot).toMatchSnapshot();
  });

  it('the Quote component renders correctly:', () => {
    const quoteSnapshot = renderer
      .create(
        <Quote />,
      )
      .toJSON();
    expect(quoteSnapshot).toMatchSnapshot();
  });

  it('the Button component renders correctly:', () => {
    const handleClick = jest.fn();
    const btnSnapshot = renderer
      .create(<Button
        name="Test button"
        color="test-color-css-class"
        wide="test-width-css-class"
        clickHandler={handleClick}
      />)
      .toJSON();
    expect(btnSnapshot).toMatchSnapshot();
  });

  it('the Display component renders correctly:', () => {
    const { next, total, operation } = { next: 'Test display text', total: null, operation: null };
    const dispSnapshot = renderer
      .create(<Display
        operation={operation}
        result={next || total}
      />)
      .toJSON();
    expect(dispSnapshot).toMatchSnapshot();
  });

  it('the NotFound component renders correctly:', () => {
    const notFoundSnapshot = renderer
      .create(
        <Router>
          <NotFound />
        </Router>,
      )
      .toJSON();
    expect(notFoundSnapshot).toMatchSnapshot();
  });
});
