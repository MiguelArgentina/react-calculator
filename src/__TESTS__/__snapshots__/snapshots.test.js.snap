// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests snapshots of components the Button com\`ponent renders correctly: 1`] = `
<button
  className="test-color-css-class test-width-css-class"
  onClick={[Function]}
  type="button"
>
  Test button
</button>
`;

exports[`Tests snapshots of components the Button component renders correctly: 1`] = `
<button
  className="test-color-css-class test-width-css-class"
  onClick={[Function]}
  type="button"
>
  Test button
</button>
`;

exports[`Tests snapshots of components the Calculator component renders correctly: 1`] = `
<div
  className="mt-4d-block d-md-flex justify-content-between"
>
  <div
    className="calc-text my-3"
  >
    Let´s do some math!
  </div>
  <div
    className="calc-wrapper"
  >
    <div
      className="display w-100"
    >
      0
    </div>
    <div
      className="button-panel-wrapper"
    >
      <div
        className="btn-row group-1"
      >
        <button
          className="btn-oth not-btn-eq"
          onClick={[Function]}
          type="button"
        >
          AC
        </button>
        <button
          className="btn-oth not-btn-eq"
          onClick={[Function]}
          type="button"
        >
          +/-
        </button>
        <button
          className="btn-oth not-btn-eq"
          onClick={[Function]}
          type="button"
        >
          %
        </button>
        <button
          className="btn-ope not-btn-eq"
          onClick={[Function]}
          type="button"
        >
          ÷
        </button>
      </div>
      <div
        className="btn-row group-2"
      >
        <button
          className="btn-oth not-btn-eq"
          onClick={[Function]}
          type="button"
        >
          7
        </button>
        <button
          className="btn-oth not-btn-eq"
          onClick={[Function]}
          type="button"
        >
          8
        </button>
        <button
          className="btn-oth not-btn-eq"
          onClick={[Function]}
          type="button"
        >
          9
        </button>
        <button
          className="btn-ope not-btn-eq"
          onClick={[Function]}
          type="button"
        >
          x
        </button>
      </div>
      <div
        className="btn-row group-3"
      >
        <button
          className="btn-oth not-btn-eq"
          onClick={[Function]}
          type="button"
        >
          4
        </button>
        <button
          className="btn-oth not-btn-eq"
          onClick={[Function]}
          type="button"
        >
          5
        </button>
        <button
          className="btn-oth not-btn-eq"
          onClick={[Function]}
          type="button"
        >
          6
        </button>
        <button
          className="btn-ope not-btn-eq"
          onClick={[Function]}
          type="button"
        >
          -
        </button>
      </div>
      <div
        className="btn-row group-4"
      >
        <button
          className="btn-oth not-btn-eq"
          onClick={[Function]}
          type="button"
        >
          1
        </button>
        <button
          className="btn-oth not-btn-eq"
          onClick={[Function]}
          type="button"
        >
          2
        </button>
        <button
          className="btn-oth not-btn-eq"
          onClick={[Function]}
          type="button"
        >
          3
        </button>
        <button
          className="btn-ope not-btn-eq"
          onClick={[Function]}
          type="button"
        >
          +
        </button>
      </div>
      <div
        className="btn-row group-5"
      >
        <button
          className="btn-oth not-btn-eq"
          onClick={[Function]}
          type="button"
        >
          0
        </button>
        <button
          className="btn-oth not-btn-eq"
          onClick={[Function]}
          type="button"
        >
          .
        </button>
        <button
          className="btn-oth btn-eq"
          onClick={[Function]}
          type="button"
        >
          =
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Tests snapshots of components the Display component renders correctly: 1`] = `
<div
  className="display w-100"
>
  Test display text
</div>
`;

exports[`Tests snapshots of components the Home component renders correctly: 1`] = `
<div>
  <h1
    className="text-start ms-4 mt-4 home-title"
  >
     Welcome to our page!
  </h1>
  <p
    className="text-start ms-4 mt-4 home-para"
  >
    Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui o!cia deserunt mollit anim id est laborum. Ipsu
  </p>
  <p
    className="text-start ms-4 mt-4 home-para"
  >
    Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui o!cia deserunt mollit anim id est laborum. Ipsu
  </p>
</div>
`;

exports[`Tests snapshots of components the NotFound component renders correctly: 1`] = `
Array [
  <div
    className="text-start home-para quote-para"
  >
    The page you are looking for does not exist.
  </div>,
  <p>
    <a
      className="back-home-link"
      href="/"
      onClick={[Function]}
    >
      Back to home
    </a>
  </p>,
]
`;

exports[`Tests snapshots of components the NotFound components renders correctly: 1`] = `
Array [
  <div
    className="text-start home-para quote-para"
  >
    The page you are looking for does not exist.
  </div>,
  <p>
    <a
      className="back-home-link"
      href="/"
      onClick={[Function]}
    >
      Back to home
    </a>
  </p>,
]
`;

exports[`Tests snapshots of components the Quote component renders correctly: 1`] = `
<div>
  <p
    className="text-start home-para quote-para"
  >
    Loading...
  </p>
</div>
`;
